src/app/core/public/api/impl/Port.ts

```diff
@@ -34,6 +35,24 @@ export class PortImpl<
         return this.getObj().index;
     }
 
+    public get originPos(): Vector {
+        // TODO[model_refactor_api]: This probably needs to be calculated explicitly here ?
```

Comment:
Yeah, what if the assembler hasn't updated it yet after a change, or if it was just created?
Especially since (i think?) we wanted to move the assembling to happen at the beginning of the draw call or something so that if things were changed multiple times in succession we don't re-assemble multiple times needlessly?
I'll add to discussion notes too

Reply:
We should `dirty` the prims that are changed, and then when we need information from them, we can re-compute at that point